index.bs

```diff
@@ -1691,11 +1691,10 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
         ::  [=set/For each=] |authenticator| in |issuedRequests| invoke the [=authenticatorCancel=] operation on |authenticator|
             and [=set/remove=] |authenticator| from |issuedRequests|. Return a {{DOMException}} whose name is "{{NotAllowedError}}".
 
-        :   If the <code>|options|.{{CredentialCreationOptions/signal}}</code> is present and its
-            [=AbortSignal/aborted flag=] is set to [TRUE],
+        :   If the <code>|options|.{{CredentialCreationOptions/signal}}</code> is present and [=AbortSignal/aborted=],
         ::  [=set/For each=] |authenticator| in |issuedRequests| invoke the [=authenticatorCancel=]
-            operation on |authenticator| and [=set/remove=] |authenticator| from |issuedRequests|. Then return a {{DOMException}}
-            whose name is "{{AbortError}}" and terminate this algorithm.
+            operation on |authenticator| and [=set/remove=] |authenticator| from |issuedRequests|. Then return the
+            <code>|options|.{{CredentialCreationOptions/signal}}</code>'s [=AbortSignal/abort reason=] and terminate this algorithm.
```

Comment:
I'm not sure if it's possible to simply wrap the `abort reason` in an "AbortError" DOMException object for example. That's why I'm thinking maybe we could check if the `abort reason` is an exception first, and return it if that is the case, and if not, we could return an "AbortError" DOMException like before. Does that sound reasonable?

Reply:
I thought Step 10.2 of the [Create a Credential](https://w3c.github.io/webappsec-credential-management/#algorithm-create) algorithm already rejects with the given exception `r`, so by checking the `abort reason`, and ensuring that we always return an exception if the signal is aborted (like we already do now by returning an "AbortError" DOMException), it wouldn't break this contract. 

Maybe I'm missing some other part to the logic though? Is the concern more that other APIs return a promise rejected with the exception, rather than returning the exception itself(like it is done in this spec and the WebOTP one as far as I'm aware)?

Reply:
The problem is that if the abort reason is the string "`blah`", the promise should be rejected with that string and not with an exception object. That's what we do for other APIs, right?

Reply:
Oops, I meant to write "throw or return". :-)

Reply:
Ah yes, thank you for the explanation! I've updated this PR to throw instead of return and done the same in the WebOTP PR (https://github.com/WICG/web-otp/pull/57) as well. I'm currently working on creating a PR for the Credential Management spec as well with updates related to this behavior change, and hope to have that up soon as well. I will update here when that is done.

Reply:
Note: Credman issue w3c/webappsec-credential-management#148 might also figure into things here.

Reply:
FYI, here is the Credential Management spec PR: https://github.com/w3c/webappsec-credential-management/pull/196